components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1953,6 +2098,73 @@ where
         self.fsm.hibernate_state.count_vote(from.get_id());
     }
 
+    #[inline]
+    fn check_force_leader(&mut self) {
+        if let Some(ForceLeaderState::PreForceLeader {
```

Comment:
nit;
From code readability perspective, I think this part of code can be improved if there is less nested parentheses (especially when the code inside the block is large), and make the whole procedure more clearly by moving the “core” logic (`on_enter_force_leader()`)  to the end of this function. Something like:

```
if !self.fsm.peer.force_leader.is_some() {
    return;
}
let Some(term, expected..) = ..;
if ... {
    warn!("");
    return;
} else if .. {
    warn!("");
    return;
} else {
    ...
}
self.on_enter_force_leader();
```

I know that you were trying to make the code efficient and reuse the code as much as possible. But, it might not be very friendly to a future maintainer who has no context about this feature.

I will leave this for you to decide though. Rest LGTM

Reply:
change it, PTAL again